gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientImpl.java

```diff
@@ -136,6 +146,45 @@ public WritableByteChannel create(StorageResourceId resourceId, CreateObjectOpti
     return channel;
   }
 
+  /**
+   * See {@link GoogleCloudStorage#createBucket(String, CreateBucketOptions)} for details about
+   * expected behavior.
+   */
+  @Override
+  public void createBucket(String bucketName, CreateBucketOptions options) throws IOException {
+    logger.atFiner().log("createBucket(%s)", bucketName);
+    checkArgument(!isNullOrEmpty(bucketName), "bucketName must not be null or empty");
+    checkNotNull(options, "options must not be null");
+    checkNotNull(storageOptions.getProjectId(), "projectId must not be null");
+
+    BucketInfo.Builder bucketInfoBuilder =
+        BucketInfo.newBuilder(bucketName).setLocation(options.getLocation());
+
+    if (options.getStorageClass() != null) {
+      bucketInfoBuilder.setStorageClass(
+          StorageClass.valueOfStrict(options.getStorageClass().toUpperCase()));
+    }
+    if (options.getTtl() != null) {
+      bucketInfoBuilder.setLifecycleRules(
+          Collections.singletonList(
+              new BucketInfo.LifecycleRule(
+                  LifecycleAction.newDeleteAction(),
+                  LifecycleCondition.newBuilder()
+                      .setAge(toIntExact(options.getTtl().toDays()))
+                      .build())));
+    }
+    try {
+      storage.create(bucketInfoBuilder.build());
+    } catch (StorageException e) {
+      if (errorExtractor.getErrorType(e) == ErrorType.ALREADY_EXISTS) {
+        throw (FileAlreadyExistsException)
```

Comment:
why the type cast is required?

Reply:
Discussed offline, this is required as the method is expected to throw IO exception only.